api/go.mod

```diff
@@ -0,0 +1,22 @@
+module github.com/containerd/containerd/api
+
+go 1.22.0
+
+require (
+	github.com/containerd/ttrpc v1.2.3
+	github.com/containerd/typeurl/v2 v2.1.1
+	github.com/opencontainers/image-spec v1.1.0
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240401170217-c3f982113cda
```

Comment:
The verions of dependencies in the root go.mod and in the api should be same so it dont cause conflicts later. Currently its at `google.golang.org/genproto/googleapis/rpc v0.0.0-20240415180920-8c6c420018be`  https://github.com/containerd/containerd/pull/10151/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R74

Reply:
This will not change the version in the main repository, the go.mod only needs to specify a minimum version. Since API isn't compiled directly, the version isn't too important here. I think it would only be an issue for API to be ahead of the main repository, but also don't want it to be ahead of 1.6/1.7 since we intend to backport there.

Reply:
agree with @dmcgowan here